Makefile

```diff
@@ -66,10 +66,28 @@ ifeq ($(HELM),$(TMPDIR)/helm)
 	curl -Ls https://get.helm.sh/helm-$(HELM_VERSION)-linux-amd64.tar.gz | tar xz -C $(TMPDIR) --strip-components=1 'linux-amd64/helm'
 endif
 
+# Deterministically create helm packages by:
+#
+# - Using `helm package` to create the initial package (useful for including chart dependencies
+#   properly)
+# - Untarring the package
+# - Recreate the package using `tar`, speficying time of the last git commit to the package
+#   source as the files' mtime, as well as ordering files deterministically, meaning that unchanged
+#   content will result in the same output package.
 $(STABLE_TARGETS) $(STAGING_TARGETS): $$(wildcard $$(patsubst docs/%.tgz,%/*,$$@)) $$(wildcard $$(patsubst docs/%.tgz,%/*/*,$$@))
 $(STABLE_TARGETS) $(STAGING_TARGETS): $(TMPDIR)/.helm/repository/local/index.yaml
 	@mkdir -p $(shell dirname $@)
-	$(HELM) --home $(TMPDIR)/.helm package $(shell echo $@ | sed 's@docs/\(.*\)-[v0-9][0-9.]*.tgz@\1@') -d $(shell dirname $@)
+	$(eval PACKAGE_SRC := $(shell echo $@ | sed 's@docs/\(.*\)-[v0-9][0-9.]*.tgz@\1@'))
```

Comment:
Not a blocker but it can match invalid versions like v0.1.1.1.1.

Reply:
Yeah, and that's OK because practically, that ain't going to happen.